incompatible with FTB Quests - Crash if using machines

e.g. putting stema and a ingot in the bronze compressor cause a crash if FTB Mods are installed.

if I set enableFtbQuestsIntegration to false it works.
```
---- Minecraft Crash Report ----
// But it works on my machine.

Time: 2024-04-12 08:52:09
Description: Ticking block entity

java.lang.ExceptionInInitializerError: null
	at aztech.modern_industrialization.stats.PlayerStatistics.addProducedItems(PlayerStatistics.java:100) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	at aztech.modern_industrialization.machines.components.CrafterComponent.putItemOutputs(CrafterComponent.java:526) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	at aztech.modern_industrialization.machines.components.CrafterComponent.tickRecipe(CrafterComponent.java:226) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	at aztech.modern_industrialization.machines.blockentities.AbstractCraftingMachineBlockEntity.tick(AbstractCraftingMachineBlockEntity.java:91) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	at aztech.modern_industrialization.machines.blockentities.SteamCraftingMachineBlockEntity.tick(SteamCraftingMachineBlockEntity.java:92) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	at aztech.modern_industrialization.blocks.TickableBlock.lambda$getTicker$0(TickableBlock.java:45) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	at net.minecraft.world.level.chunk.LevelChunk$BoundTickingBlockEntity.tick(LevelChunk.java:691) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading}
	at net.minecraft.world.level.chunk.LevelChunk$RebindableTickingBlockEntityWrapper.tick(LevelChunk.java:771) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading}
	at net.minecraft.world.level.Level.tickBlockEntities(Level.java:545) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.level.ServerLevel.tick(ServerLevel.java:409) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:969) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:855) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.client.server.IntegratedServer.tickServer(IntegratedServer.java:106) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:700) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:255) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:accesstransformer:B}
	at java.lang.Thread.run(Thread.java:833) ~[?:?] {}
Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: aztech.modern_industrialization.compat.ftbquests.FTBQuestsFacadeImpl
	at aztech.modern_industrialization.compat.ftbquests.FTBQuestsFacade.getInstance(FTBQuestsFacade.java:40) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	at aztech.modern_industrialization.compat.ftbquests.FTBQuestsFacade.<clinit>(FTBQuestsFacade.java:32) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	... 16 more
Caused by: java.lang.ClassNotFoundException: aztech.modern_industrialization.compat.ftbquests.FTBQuestsFacadeImpl
	at cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:197) ~[securejarhandler-2.1.24.jar:?] {}
	at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?] {}
	at java.lang.Class.forName0(Native Method) ~[?:?] {}
	at java.lang.Class.forName(Class.java:375) ~[?:?] {}
	at aztech.modern_industrialization.compat.ftbquests.FTBQuestsFacade.getInstance(FTBQuestsFacade.java:37) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	at aztech.modern_industrialization.compat.ftbquests.FTBQuestsFacade.<clinit>(FTBQuestsFacade.java:32) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	... 16 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at aztech.modern_industrialization.stats.PlayerStatistics.addProducedItems(PlayerStatistics.java:100) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	at aztech.modern_industrialization.machines.components.CrafterComponent.putItemOutputs(CrafterComponent.java:526) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	at aztech.modern_industrialization.machines.components.CrafterComponent.tickRecipe(CrafterComponent.java:226) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	at aztech.modern_industrialization.machines.blockentities.AbstractCraftingMachineBlockEntity.tick(AbstractCraftingMachineBlockEntity.java:91) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	at aztech.modern_industrialization.machines.blockentities.SteamCraftingMachineBlockEntity.tick(SteamCraftingMachineBlockEntity.java:92) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	at aztech.modern_industrialization.blocks.TickableBlock.lambda$getTicker$0(TickableBlock.java:45) ~[Modern-Industrialization-2.0.17-alpha-all.jar%23184!/:?] {re:classloading}
	at net.minecraft.world.level.chunk.LevelChunk$BoundTickingBlockEntity.tick(LevelChunk.java:691) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading}
	at net.minecraft.world.level.chunk.LevelChunk$RebindableTickingBlockEntityWrapper.tick(LevelChunk.java:771) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading}
	at net.minecraft.world.level.Level.tickBlockEntities(Level.java:545) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
-- Block entity being ticked --
Details:
	Name: modern_industrialization:bronze_compressor // aztech.modern_industrialization.machines.blockentities.SteamCraftingMachineBlockEntity
	Block: Block{modern_industrialization:bronze_compressor}
	Block location: World: (18,71,-37), Section: (at 2,7,11 in 1,4,-3; chunk contains blocks 16,-64,-48 to 31,319,-33), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,-64,-512 to 511,319,-1)
	Block: Block{modern_industrialization:bronze_compressor}
	Block location: World: (18,71,-37), Section: (at 2,7,11 in 1,4,-3; chunk contains blocks 16,-64,-48 to 31,319,-33), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,-64,-512 to 511,319,-1)
Stacktrace:
	at net.minecraft.world.level.chunk.LevelChunk$BoundTickingBlockEntity.tick(LevelChunk.java:691) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading}
	at net.minecraft.world.level.chunk.LevelChunk$RebindableTickingBlockEntityWrapper.tick(LevelChunk.java:771) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading}
	at net.minecraft.world.level.Level.tickBlockEntities(Level.java:545) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.level.ServerLevel.tick(ServerLevel.java:409) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:969) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:855) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.client.server.IntegratedServer.tickServer(IntegratedServer.java:106) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:700) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:255) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:accesstransformer:B}
	at java.lang.Thread.run(Thread.java:833) ~[?:?] {}


-- Affected level --
Details:
	All players: 0 total; []
	Chunk stats: 2209
	Level dimension: minecraft:overworld
	Level spawn location: World: (16,71,-48), Section: (at 0,7,0 in 1,4,-3; chunk contains blocks 16,-64,-48 to 31,319,-33), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,-64,-512 to 511,319,-1)
	Level time: 44743 game time, 7707 day time
	Level name: xx
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: true
	Level weather: Rain time: 3456 (now: false), thunder time: 117011 (now: false)
	Known server brands: neoforge
	Removed feature flags: 
	Level was modded: true
	Level storage version: 0x04ABD - Anvil
	Loaded entity count: 138
Stacktrace:
	at net.minecraft.server.level.ServerLevel.fillReportDetails(ServerLevel.java:1695) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:972) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:855) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.client.server.IntegratedServer.tickServer(IntegratedServer.java:106) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:700) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:255) ~[client-1.20.4-20231207.154220-srg.jar%23185!/:?] {re:classloading,pl:accesstransformer:B}
	at java.lang.Thread.run(Thread.java:833) ~[?:?] {}


-- System Details --
Details:
	Minecraft Version: 1.20.4
	Minecraft Version ID: 1.20.4
	Operating System: Linux (amd64) version 6.7.7-1-default
	Java Version: 17.0.8, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 1306322448 bytes (1245 MiB) / 2327838720 bytes (2220 MiB) up to 4294967296 bytes (4096 MiB)
	CPUs: 16
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Ryzen 7 2700X Eight-Core Processor
	Identifier: AuthenticAMD Family 23 Model 8 Stepping 2
	Microarchitecture: Zen+
	Frequency (GHz): -0.00
	Number of physical packages: 1
	Number of physical CPUs: 8
	Number of logical CPUs: 16
	Graphics card #0 name: Ellesmere [Radeon RX 470/480/570/570X/580/580X/590]
	Graphics card #0 vendor: Advanced Micro Devices, Inc. [AMD/ATI] (0x1002)
	Graphics card #0 VRAM (MB): 258.00
	Graphics card #0 deviceId: 0x67df
	Graphics card #0 versionInfo: unknown
	Virtual memory max (MB): 22060.05
	Virtual memory used (MB): 12323.26
	Swap memory total (MB): 2048.00
	Swap memory used (MB): 0.00
	JVM Flags: 3 total; -Xss1M -Xmx4096m -Xms256m
	Server Running: true
	Player Count: 0 / 8; []
	Data Packs: vanilla, mod_data, mod:ftbultimine (incompatible), mod:architectury (incompatible), mod:ftblibrary (incompatible), mod:jei (incompatible), mod:grandpower, mod:ftbteams (incompatible), mod:ftbquests (incompatible), mod:ftbchunks (incompatible), mod:cloth_config (incompatible), mod:patchouli (incompatible), mod:neoforge, mod:modern_industrialization
	Enabled Feature Flags: minecraft:update_1_21, minecraft:bundle, minecraft:vanilla
	World Generation: Stable
	Type: Integrated Server (map_client.txt)
	Is Modded: Definitely; Client brand changed to 'neoforge'; Server brand changed to 'neoforge'
	Launched Version: neoforge-20.4.225
	ModLauncher: 10.0.9+10.0.9+main.dcd20f30
	ModLauncher launch target: forgeclient
	ModLauncher naming: srg
	ModLauncher services: 
		sponge-mixin-0.13.1+mixin.0.8.5.jar mixin PLUGINSERVICE 
		loader-2.0.17.jar slf4jfixer PLUGINSERVICE 
		loader-2.0.17.jar runtime_enum_extender PLUGINSERVICE 
		at-modlauncher-10.0.1.jar accesstransformer PLUGINSERVICE 
		loader-2.0.17.jar runtimedistcleaner PLUGINSERVICE 
		modlauncher-10.0.9.jar mixin TRANSFORMATIONSERVICE 
		modlauncher-10.0.9.jar mixin-synthetic-package TRANSFORMATIONSERVICE 
		modlauncher-10.0.9.jar fml TRANSFORMATIONSERVICE 
	FML Language Providers: 
		javafml@2.0
		lowcodefml@2.0
		minecraft@2.0
	Mod List: 
		client-1.20.4-20231207.154220-srg.jar             |Minecraft                     |minecraft                     |1.20.4              |DONE      |Manifest: a1:d4:5e:04:4f:d3:d6:e0:7b:37:97:cf:77:b0:de:ad:4a:47:ce:8c:96:49:5f:0a:cf:8c:ae:b2:6d:4b:8a:3f
		ftb-ultimine-neoforge-2004.1.0.jar                |FTB Ultimine                  |ftbultimine                   |2004.1.0            |DONE      |Manifest: NOSIGNATURE
		architectury-11.1.17-neoforge.jar                 |Architectury                  |architectury                  |11.1.17             |DONE      |Manifest: NOSIGNATURE
		ftb-library-neoforge-2004.2.1.jar                 |FTB Library                   |ftblibrary                    |2004.2.1            |DONE      |Manifest: NOSIGNATURE
		jei-1.20.4-neoforge-17.3.0.49.jar                 |Just Enough Items             |jei                           |17.3.0.49           |DONE      |Manifest: NOSIGNATURE
		GrandPower-1.0.0.jar                              |Grand Power                   |grandpower                    |1.0.0               |DONE      |Manifest: NOSIGNATURE
		ftb-teams-neoforge-2004.1.2.jar                   |FTB Teams                     |ftbteams                      |2004.1.2            |DONE      |Manifest: NOSIGNATURE
		ftb-quests-neoforge-2004.2.0.jar                  |FTB Quests                    |ftbquests                     |2004.2.0            |DONE      |Manifest: NOSIGNATURE
		ftb-chunks-neoforge-2004.1.2.jar                  |FTB Chunks                    |ftbchunks                     |2004.1.2            |DONE      |Manifest: NOSIGNATURE
		cloth-config-13.0.121-neoforge.jar                |Cloth Config v13 API          |cloth_config                  |13.0.121            |DONE      |Manifest: NOSIGNATURE
		Patchouli-1.20.4-85-NEOFORGE-SNAPSHOT.jar         |Patchouli                     |patchouli                     |1.20.4-85-NEOFORGE-S|DONE      |Manifest: NOSIGNATURE
		neoforge-20.4.225-universal.jar                   |NeoForge                      |neoforge                      |20.4.225            |DONE      |Manifest: NOSIGNATURE
		Modern-Industrialization-2.0.17-alpha-all.jar     |Modern Industrialization      |modern_industrialization      |2.0.17-alpha        |DONE      |Manifest: NOSIGNATURE
	Crash Report UUID: bbe88c01-2d83-47ca-8697-6b185279783f
	FML: 20.4
	NeoForge: net.neoforged:20.4.225
```

--- a/build.gradle
+++ b/build.gradle
@@ -230,11 +230,7 @@ dependencies {
     implementation "earth.terrarium.argonauts:argonauts-neoforge-${project.minecraft_version}:${project.argonauts_version}"
     implementation "earth.terrarium.athena:athena-neoforge-${project.minecraft_version}:${project.athena_version}"
 
-//    // TODO 1.20: FTB stuff
-////    modCompileOnly("dev.ftb.mods:ftb-quests-fabric:${project.ftb_quests_version}") {
-////        exclude(group: "net.fabricmc.fabric-api")
-////        exclude(group: "me.shedaniel")
-////    }
+    compileOnly "dev.ftb.mods:ftb-quests-neoforge:${project.ftb_quests_version}"
     compileOnly "dev.ftb.mods:ftb-teams-neoforge:${project.ftb_teams_version}"
 }
 

--- a/gradle.properties
+++ b/gradle.properties
@@ -35,6 +35,7 @@ argonauts_version=1.2.0
 athena_version=3.3.2
 emi_minecraft_version=1.20.4
 emi_version=1.1.0
+ftb_quests_version=2004.2.0-SNAPSHOT
 ftb_teams_version=2004.1.2-SNAPSHOT
 jade_file_id=5109393
 jei_minecraft_version=1.20.4
@@ -43,8 +44,5 @@ patchouli_version=1.20.4-85-NEOFORGE-SNAPSHOT
 rei_version=14.0.688
 rei_annotations_version=12.0.93
 
-#	ae2_version=15.0.7-beta
-#	ftb_quests_version=1902.4.16-build.235
-#	ftb_teams_version=2001.1.4
 #	kubejs_version=2001.6.2-build.4
 #

--- a/src/main/java/aztech/modern_industrialization/compat/ftbquests/FTBQuestsFacadeImpl.java
+++ b/src/main/java/aztech/modern_industrialization/compat/ftbquests/FTBQuestsFacadeImpl.java
@@ -23,32 +23,40 @@
  */
 package aztech.modern_industrialization.compat.ftbquests;
 
+import dev.ftb.mods.ftbquests.item.MissingItem;
+import dev.ftb.mods.ftbquests.quest.ServerQuestFile;
+import dev.ftb.mods.ftbquests.quest.task.ItemTask;
+import dev.ftb.mods.ftbteams.api.FTBTeamsAPI;
 import java.util.UUID;
 import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
 
 public class FTBQuestsFacadeImpl implements FTBQuestsFacade {
     @Override
     public void addCompleted(UUID uuid, Item item, long amount) {
-        // TODO 1.20: wait for FTB Quests update
-//        var file = ServerQuestFile.INSTANCE;
-//        var data = file.getNullableTeamData(FTBTeamsAPI.getPlayerTeamID(uuid));
-//
-//        if (data == null || data.isLocked()) {
-//            return;
-//        }
-//
-//        ItemStack stack = new ItemStack(item, (int) amount);
-//
-//        for (var task : file.getSubmitTasks()) {
-//            if (task instanceof ItemTask itemTask && data.canStartTasks(task.quest)) {
-//                if (data.isCompleted(task) || itemTask.item.getItem() instanceof MissingItem || item instanceof MissingItem) {
-//                    continue;
-//                }
-//
-//                if (!task.consumesResources() && itemTask.test(stack)) {
-//                    data.addProgress(task, amount);
-//                }
-//            }
-//        }
+        var file = ServerQuestFile.INSTANCE;
+        var team = FTBTeamsAPI.api().getManager().getTeamForPlayerID(uuid).orElse(null);
+        if (team == null) {
+            return;
+        }
+        var data = file.getNullableTeamData(team.getId());
+
+        if (data == null || data.isLocked()) {
+            return;
+        }
+
+        ItemStack stack = new ItemStack(item, (int) amount);
+
+        for (var task : file.getSubmitTasks()) {
+            if (task instanceof ItemTask itemTask && data.canStartTasks(task.getQuest())) {
+                if (data.isCompleted(task) || itemTask.getItemStack().getItem() instanceof MissingItem || item instanceof MissingItem) {
+                    continue;
+                }
+
+                if (!task.consumesResources() && itemTask.test(stack)) {
+                    data.addProgress(task, amount);
+                }
+            }
+        }
     }
 }

